Content.Server/GameTicking/Rules/ThiefRuleSystem.cs

```diff
@@ -89,11 +89,42 @@ private void DoThiefStart(ThiefRuleComponent component)
         while (component.ThievesMinds.Count < numberOfThievesToSelect && thiefPool.Count > 0)
         {
             Log.Info($"{numberOfThievesToSelect} thieves required, {component.ThievesMinds.Count} currently chosen, {thiefPool.Count} potentials");
-            var selectedThieves = _antagSelection.PickAntag(numberOfThievesToSelect - component.ThievesMinds.Count, thiefPool);
-            foreach (var thief in selectedThieves)
+
+            // check if there aren't enough players available for a thief round-start
+            if (thiefPool.Count >= numberOfThievesToSelect - component.ThievesMinds.Count)
+            {
+                var selectedThieves = _antagSelection.PickAntag(numberOfThievesToSelect - component.ThievesMinds.Count, thiefPool);
+
+                foreach (var thief in selectedThieves)
+                {
+                    MakeThief(component, thief, component.PacifistThieves);
+                }
+            }
+            else
             {
-                MakeThief(component, thief, component.PacifistThieves);
+                Log.Info("Not enough potential thieves available.");
+
+                // determine the maximum number of thieves that can be picked from the available pool, without causing an ArgumentOutOfRangeException.
+                var maxThievesToPick = Math.Min(thiefPool.Count, numberOfThievesToSelect - component.ThievesMinds.Count);
+
+                if (maxThievesToPick > 0)
+                {
+                    // then pick as many thieves as *safely* possible, error handling for testing purposes
+                    var selectedThieves = _antagSelection.PickAntag(maxThievesToPick, thiefPool);
+
+                    foreach (var thief in selectedThieves)
+                    {
+                        MakeThief(component, thief, component.PacifistThieves);
+                    }
+
+                }
+                else
+                {
+                    Log.Info("No potential thieves available.");
+                }
             }
+
+
         }
```

Comment:
Is it possible to just share this code with nukies / traitors instead as every new antag we add is going to need to implement this.

Reply:
If this is the case why not implement it in the already ongoing antag refactor?

Reply:
it already is implemented there i think

Reply:
uuh possible yes, but i'd have to learn a new system. right now this kinda bug only affects thief, the game knows how to handle any other antag in this situation. want me to do it anyways?